### ISSUE TITLE
Private-preserving classification

### ISSUE THREAD

**Issue by 9310gaurav** [inspire-group/RON-Gauss#1]
I have a question regarding the classifier trained on the synthetic data. To test the quality of the synthetic data, you use a classifier trained on it and test it's performance on the real test set. But do you perform the pre-processing and projection on the test set as well? If you do then how is data centering done? And if not then how do you ensure the learned classifier will work on unnormalized data if it's trained on normalized and projected data?

**Comment by tchanyaswad** [comment]
The test data are also centered and normalized. The centering is done by using the DP-protected mean derived from the training data. The normalization is done sample-wise, so it can be performed on the test data without touching the training data.

**Comment by 9310gaurav** [comment]
Thanks a lot @tchanyaswad for the quick response. I was able to run this on Adult dataset and it performs really well compared to other heavyweight methods like DP-GAN which take a lot of time to train as well.
